lib/json_api_controller/precondition_check.rb

```diff
@@ -0,0 +1,28 @@
+module JsonApiController
+  module PreconditionCheck
+    def precondition_check
+      if !precondition
+        raise PreconditionNotPresent, precondition_error_msg
+      elsif precondition_fails?
+        raise PreconditionFailed
+      end
+    end
+    
+    def precondition
+      case action_name
+      when "show"
```

Comment:
Sorry, I meant are we using the show action when checking the header preconditions? It seemed like we are only checking preconditions on the update / delete actions, so the show is not needed.